src/utils/conversion.ts

```diff
@@ -687,42 +723,153 @@ export function permissionsToMeshPermissions(
 ): MeshPermissions {
   const { tokens, transactions, portfolios } = permissions;
 
-  const extrinsicDict: Record<string, string[]> = {};
-  // eslint-disable-next-line @typescript-eslint/naming-convention
-  let extrinsic: { pallet_name: string; dispatchable_names: string[] }[] | null = null;
+  const extrinsicDict: Record<string, { tx: string[]; exception?: true } | null> = {};
+  let extrinsic: PermissionsEnum<PalletPermissions> = 'Whole';
 
   if (transactions) {
-    uniq(transactions)
+    const message =
+      'Attempting to add permissions for specific transactions as well as the entire module';
+    const { values: txValues, exceptions = [], type } = transactions;
+    uniq(txValues)
       .sort()
       .forEach(tag => {
-        const [modName, txName] = tag.split('.');
-
-        const palletName = stringUpperFirst(modName);
-        const dispatchableName = snakeCase(txName);
+        if (tag.includes('.')) {
+          const { palletName, dispatchableName } = splitTag(tag as TxTag);
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          let pallet = extrinsicDict[palletName];
+
+          if (pallet === null) {
+            throw new PolymeshError({
+              code: ErrorCode.ValidationError,
+              message,
+              data: {
+                module: palletName,
+                transactions: [dispatchableName],
+              },
+            });
+          } else if (pallet === undefined) {
+            pallet = extrinsicDict[palletName] = { tx: [] };
+          }
 
-        const pallet = (extrinsicDict[palletName] = extrinsicDict[palletName] || []);
+          pallet.tx.push(dispatchableName);
+        } else {
+          const pallet = extrinsicDict[tag];
+
+          if (pallet) {
```

Comment:
`pallet` never will defined at this point because the only way to accomplish that is removing the sort()